src/StateLabel.js

```diff
@@ -1,65 +1,86 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 import styled from 'styled-components'
-import {GitMerge, GitPullRequest, IssueClosed, IssueOpened} from '@primer/octicons-react'
-import theme, {colors} from './theme'
+import {GitMerge, GitPullRequest, IssueClosed, IssueOpened, Question} from '@primer/octicons-react'
+import {variant} from 'styled-system'
+import theme from './theme'
 import {COMMON, get} from './constants'
 import StyledOcticon from './StyledOcticon'
 import sx from './sx'
-
-const statusMap = {
-  issueClosed: colors.red[6],
-  pullClosed: colors.red[5],
-  pullMerged: colors.purple[5],
-  issueOpened: '#159739', // Custom green
-  pullOpened: '#2cbe4e', // This was generated by a sass function in Primer, so using a hex here
-  gray: colors.gray[5]
-}
+import {useDeprecation} from './utils/deprecate'
 
 const octiconMap = {
   issueOpened: IssueOpened,
   pullOpened: GitPullRequest,
   issueClosed: IssueClosed,
   pullClosed: GitPullRequest,
-  pullMerged: GitMerge
+  pullMerged: GitMerge,
+  draft: GitPullRequest
 }
 
-function StateLabelBase({className, status, small = false, children}) {
-  const octiconProps = small ? {width: '1em'} : {}
-  return (
-    <span className={className}>
-      {status && <StyledOcticon mr={1} {...octiconProps} icon={octiconMap[status]} />}
-      {children}
-    </span>
-  )
-}
+const colorVariants = variant({
+  prop: 'status',
+  scale: 'stateLabels.status',
+  // https://styled-system.com/variants/#migrating-from-legacy-api
+  variants: {
+    ...theme.stateLabels.status
+  }
+})
+
+const sizeVariants = variant({
+  prop: 'variant',
+  scale: 'stateLabels.sizes',
+  // https://styled-system.com/variants/#migrating-from-legacy-api
+  variants: {
+    ...theme.stateLabels.sizes
+  }
+})
```

Comment:
if you reference the colors directly as I mentioned in my comments in theme.js you can remove the variants key from these objects (I recently learned this from @colebemis!):

```suggestion
const colorVariants = variant({
  prop: 'status',
  scale: 'stateLabels.status',
})

const sizeVariants = variant({
  prop: 'variant',
  scale: 'stateLabels.sizes'
})
```